tom256/src/lib.rs

```diff
@@ -28,18 +38,28 @@ pub fn generate_proof(input: JsValue, ring: JsValue) -> Result<JsValue, JsValue>
 
     let zk_attest_proof = ZkAttestProof::construct(&mut rng, pedersen, input, &ring)?;
 
-    JsValue::from_serde(&zk_attest_proof).map_err(|e| JsValue::from(e.to_string()))
+    let proof_binary = zk_attest_proof
+        .try_to_vec()
+        .map_err(|e| JsValue::from(e.to_string()))?;
+
+    let proof_output = ProofOutput {
+        guild_id: zk_attest_proof.guild_id,
+        r_point: zk_attest_proof.r_point,
+        proof_binary,
```

Comment:
do we double serialize `guild_id` and `r_point` here? Is there a better way to do this and not include those two fields in in the binary as well?

Reply:
But probably it's fine because _hopefully_ those two extra fields do not add too much to the proof size.

Reply:
The size of the binary proof is around 160 KB, so adding another point and a string does not increase it substantially but it is more simple. Also, the interface might still frequently change so I do not think we should overspecialize for the current iteration.